ledger/src/shred.rs

```diff
@@ -467,8 +473,33 @@ impl Shred {
         self.common_header.signature
     }
 
-    pub fn seed(&self) -> [u8; 32] {
+    pub fn seed(
+        &self,
+        leader_schedule_cache: &Arc<LeaderScheduleCache>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> [u8; 32] {
         let mut seed = [0; 32];
+
+        {
+            let r_bf = bank_forks.read().unwrap();
+            let root_bank = r_bf.root_bank();
+            let maybe_slot_bank = r_bf.get(self.slot());
```

Comment:
This is holding to the bank-forks lock longer than it should. Instead you can do:
```
let (root_bank, slot_bank) = {
    let bank_forks = bank_forks.read().unwrap();
    (bank_forks.root_bank(), bank_forks.get(self.slot()).cloned())
}
```